### PR TITLE
PVS-Studio Static Code Analyzer support

### PR DESCRIPTION
**Pull request by feeelin**
Hello! We added support for JSON report of PVS-Studio static analyzer in report-converter so that you could view the analyzer's report in CodeChecker!

Usage:
`report-converter -t pvs-studio -o <path to converted reports> <path to PVS-Studio's report.json>`

### CODE REVIEW

**Review by vodorok — CHANGES_REQUESTED** [review]
Hi @feeelin,

Thanks for the contribution! 
Please add tests for your newly implemented report converter.

Look for already existing converter tests under: `codechecker/tools/report-converter/tests/unit/analyzers`.
You should create a separate `pvs_studio_output_test_files` in this folder, containing the sample outputs of your analyzer, the expected `.plist` (CodeChecker) output, and the source that's the basis of the analyzer output. 
You can write your test implementation by using these sample files in `test_pvs_studio_parser.py`.

**Comment by feeelin** [comment]
I have added tests for the PVS-Studio report converter by analogy with other tools.

**Review by vodorok — CHANGES_REQUESTED** [review]
Thank you, I see a few linter errors in the parser source and the test source..

https://github.com/Ericsson/codechecker/actions/runs/11106888906/job/30856471518?pr=4356
```
pycodestyle codechecker_report_converter tests --exclude tests/unit/analyzers/pyflakes_output_test_files/files
codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py:12:80: E501 line too long (80 > 79 characters)
codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py:39:80: E501 line too long (88 > 79 characters)
codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py:53:80: E501 line too long (81 > 79 characters)
codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py:57:80: E501 line too long (86 > 79 characters)
tests/unit/analyzers/test_pvs_studio_parser.py:10:1: E302 expected 2 blank lines, found 1
tests/unit/analyzers/test_pvs_studio_parser.py:28:80: E501 line too long (116 > 79 characters)
tests/unit/analyzers/test_pvs_studio_parser.py:37:80: E501 line too long (85 > 79 characters)
tests/unit/analyzers/test_pvs_studio_parser.py:38:1: W293 blank line contains whitespace
make: *** [tests/Makefile:29: pycodestyle] Error 1
```

**Comment by feeelin** [comment]
Okay, I'll fix that in the near future!

There is one question: will the severity field in the Report class be used when displaying the report in the future? If it is, I would add the transformation of our diagnostics' levels into CodeChecker's severity format so that in the future we could determine the level of PVS-Studio diagnostics more precisely.

**Review by feeelin — COMMENTED** [review]
I fixed the crashing test. Our analyzer records the full path to the file in the report, so I made a method that will make the path to the nearby sample.cpp file up to date, don't be scared by this change :)

**Review by feeelin — COMMENTED** [review]
Changed the specifics of the tests, now only the analyzer's warning will be checked instead of the whole report, because unittest caches the unupdated path to the file in `.plist` files.

Also added PVS-Studio to the list of supported analyzers and report-converter documentation.

**Comment by feeelin** [comment]
Hello @vodorok, @bruntib, @dkrupp!
Still waiting for your response to merge the changes

**Comment by feeelin** [comment]
Hello @vodorok, @bruntib, @dkrupp!
All tests is passed, still waiting for your review :)

File: tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py
Comment (dkrupp) [inline tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py:106]: does PVS studio support relative path to the source files?
If it does, I think we could test the transform() function for .json files with relative paths.

**Review by dkrupp — CHANGES_REQUESTED** [review]
Thanks a lot for your patch!!

I have some minor remarks for the test cases.
Could you please take a look at them?

File: tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py
Comment (feeelin) [reply tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py]: Probably forgot to remove that line when I added the method to set absolute paths in the json report. Fixed it.

File: tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py
Comment (feeelin) [reply tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py:87]: All right, I added this.

File: tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py
Comment (feeelin) [reply tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py:95]: All right, I added this.

File: tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py
Comment (feeelin) [reply tools/report-converter/tests/unit/analyzers/test_pvs_studio_parser.py:113]: No, PVS-Studio uses absolute paths to source files.

**Review by dkrupp — APPROVED** [review]
LGTM

File: docs/tools/report-converter.md
Comment (bruntib) [inline docs/tools/report-converter.md:415]:
```suggestion
CodeChecker store ./codechecker_pvs_studio_reports -n pvs_studio
```

File: tools/report-converter/codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py
Comment (bruntib) [inline tools/report-converter/codechecker_report_converter/analyzers/pvs_studio/analyzer_result.py:60]: Conventionally we use `error` loglevel when CodeChecker fails to continue the execution. Would you think, a `warning` is enough?

**Review by bruntib — COMMENTED** [review]
Please, add the license header to `__init__.py`

Sorry for the slow review process. After after fixing these minor things, we can merge it. Thanks for your work!

**Comment by feeelin** [comment]
Committed the latest edits. Thanks for your help!

**Review by vodorok — APPROVED** [review]
LGTM! Thanks a lot!

**Comment by feeelin** [comment]
Hi, @bruntib, @dkrupp !
When will the changes from this branch make it into the release (pypi and github)?

**Comment by vodorok** [comment]
Hi, @feeelin, we plan our next feature release for early December, CodeChecker 6.25.0. This change will be included in that release.

**Comment by feeelin** [comment]
Hi, @bruntib, @dkrupp! 
Have the plans for the 6.25.0 release changed? And will it be released before the end of this week?

**Comment by bruntib** [comment]
Hi,
We're planning to deploy a release candidate this year. The main release will come in January.

**Pull request merged**
